101-function-app-create-dedicated/azuredeploy.json

```diff
@@ -0,0 +1,105 @@
+{
+    "$schema": "http://schemas.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "appName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the function app that you wish to create."
+            }
+        },
+        "location": {
+            "type": "string",
+            "metadata": {
+                "description": "The location to use for creating the application."
+            }
+        },
+        "hostingPlanSettings": {
+            "type": "Object",
+            "defaultValue": {
```

Comment:
Why not make all of these parameters? A complex object shows up weird in Portal in the "Deploy to Azure" UI